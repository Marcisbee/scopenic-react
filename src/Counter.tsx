import React from 'react';
import { useStore } from 'statux';

interface CounterProps {
  className?: string;
}

const Counter: React.FC<CounterProps> = () => {
  const [user, setUser] = useStore('user');
  return (
    <div>
      Hello {user ? user.name : (
        <button onClick={() => setUser({ name: 'Maria' })}>Login</button>
      )}
    </div>
  );
}

export default Counter;
