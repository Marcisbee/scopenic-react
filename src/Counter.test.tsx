import React from 'react';
import { shallow } from 'enzyme';
import Store from 'statux';
import Counter from './Counter';

jest.mock('statux');

it('renders without crashing', () => {
  const component = shallow(<Counter />);
  expect(component).toMatchSnapshot();
});

it('render user name', () => {
  const user = { name: 'John' };
  const setUser = jest.fn();
  Store.useStore = jest.fn(() => [user, setUser]);
  const component = shallow(<Counter />);
  expect(component).toMatchSnapshot();
});

it('calls setUser on click', () => {
  const user = null;
  const setUser = jest.fn();
  Store.useStore = jest.fn(() => [user, setUser]);
  const component = shallow(<Counter />);
  component.find('button').simulate('click');

  expect(setUser).toHaveBeenCalledTimes(1);
  expect(setUser).toHaveBeenCalledWith({ name: 'Maria' });
});
