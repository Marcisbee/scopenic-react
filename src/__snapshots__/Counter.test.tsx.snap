// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render user name 1`] = `
<div>
  Hello 
  John
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  Hello 
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
